compiler/rustc_lint/src/context.rs

```diff
@@ -720,6 +720,9 @@ impl<'tcx> LateContext<'tcx> {
     /// Anonymous scopes such as `extern` imports are matched with `kw::Invalid`;
     /// inherent `impl` blocks are matched with the name of the type.
     ///
+    /// Instead of using this method, it is often preferable to instead use
+    /// `rustc_diagnostic_item` or a `lang_item`, which is less prone to errors.
```

Comment:
Can you explain why diagnostic items are less prone to errors? Is it because they're not tricked by re-exporting under a different name?

Reply:
It is both harder to have a typo and resilient to refactorings. Should I mention that in the docs directly?

Reply:
That would be great :)